### PR TITLE
T32: Implement ALU (shifted register) instructions

### PR DESCRIPTION
**Pull request by merryhime**
Draft as tests pending.

Am willing to reorder the opcode table in alphabetical order instead of encoding order if preferrable.

### CODE REVIEW

**Comment by gdkchan** [comment]
I prefer alphabetical order since it's easier to find instructions by name in my opinion. And it would be consistent with the A64 and A32 tables.

File: ARMeilleure/Decoders/OpCodeTable.cs [hunk 1]
```diff
@@ -1092,8 +1111,11 @@ private static void SetT16(string encoding, InstName name, InstEmitter emitter,
 
         private static void SetT32(string encoding, InstName name, InstEmitter emitter, MakeOp makeOp)
         {
-            encoding = encoding.Substring(16) + encoding.Substring(0, 16);
-            Set(encoding, AllInstT32, new InstDescriptor(name, emitter), makeOp);
+            string reversedEncoding = encoding.Substring(16) + encoding.Substring(0, 16);
+            MakeOp reversedMakeOp =
+                (InstDescriptor inst, ulong address, int opCode)
+                    => makeOp(inst, address, (int)(((uint)opCode << 16) | ((uint)opCode >> 16)));
```
Comment (gdkchan) [inline ARMeilleure/Decoders/OpCodeTable.cs:1117]:
```suggestion
                    => makeOp(inst, address, BitOperations.RotateRight((uint)opCode, 16));
```

File: ARMeilleure/Decoders/OpCodeTable.cs [hunk 2]
```diff
@@ -1045,6 +1045,26 @@ static OpCodeTable()
             SetT16("11100xxxxxxxxxxx", InstName.B,      InstEmit32.B,       OpCodeT16BImm11.Create);
 #endregion
 
+#region "OpCode Table (AArch32, T32)"
+            // Base
+            SetT32("11101011010xxxxx0xxxxxxxxxxxxxxx", InstName.Adc,      InstEmit32.Adc,      OpCodeT32AluRsImm.Create);
```
Comment (gdkchan) [inline ARMeilleure/Decoders/OpCodeTable.cs:1050]: Did you leave the extra spacing to avoid re-aligning the entire table in the future?

**Review by gdkchan — APPROVED** [review]
lgtm, minus comment.

**Review by riperiperi — APPROVED** [review]
LGTM

**Pull request merged**